### PR TITLE
add BillingType in CarrierAccount and Rate structs.

### PR DESCRIPTION
**Pull request by jchen293**
# Description

Add `billing_type` in CarrierAccount and Rate structs

# Testing

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

**Comment by nwithan8** [comment]
@jchen293 You're going to need to re-record cassettes

**Review by nwithan8 — CHANGES_REQUESTED** [review]
Missing credit card funding and deletion tests

**Comment by jchen293** [comment]
> 

It's in here: https://github.com/EasyPost/easypost-go/blob/master/tests/credit_card_test.go

**Comment by jchen293** [comment]
> Missing credit card funding and deletion tests. I know we can't run them, but we should still add them to the test suite.

It's in here: https://github.com/EasyPost/easypost-go/blob/master/tests/credit_card_test.go

**Review by nwithan8 — APPROVED** [review]

File: rate.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ type Rate struct {
 	DeliveryDate           *time.Time `json:"delivery_date,omitempty"`
 	DeliveryDateGuaranteed bool       `json:"delivery_date_guaranteed,omitempty"`
 	EstDeliveryDays        int        `json:"est_delivery_dats,omitempty"`
+	BillingType            string     `json:"billing_type,omitempty"`
```
Comment (Justintime50) [inline rate.go:29]: Does the smartrate strict need this too?

Comment (jchen293) [reply rate.go:29]: No, it doesn't. This question was answered in Java PR: https://github.com/EasyPost/easypost-java/pull/141#discussion_r892859981

**Comment by nwithan8** [comment]
@jchen293 you'll need to rebase this and re-record some cassettes

File: tests/fixture_test.go [hunk 2]
```diff
@@ -171,11 +171,7 @@ func (fixture *Fixture) OneCallBuyShipment() *easypost.Shipment {
 // If you need to re-record cassettes, increment the date below and ensure it is one day in the future,
 // USPS only does "next-day" pickups including Saturday but not Sunday or Holidays.
 func (fixture *Fixture) BasicPickup() *easypost.Pickup {
-<<<<<<< HEAD
-	pickupDate := time.Date(2022, time.June, 27, 0, 0, 0, 0, time.UTC)
```
Comment (nwithan8) [inline tests/fixture_test.go:174]: @jchen293 I think you deleted the wrong pickupDate here, the one that's left is in the past and is probably causing the cassettes/tests to fail

**Review by nwithan8 — CHANGES_REQUESTED** [review]

File: tests/fixture_test.go (hunk 2 above)
Comment (nwithan8) [reply tests/fixture_test.go:174]: Well, the new cassettes passed, somehow...

Comment (jchen293) [reply tests/fixture_test.go:174]: I forgot to save the files after rebasing before pushing the code which was causing the CI build to fail.

**Review by nwithan8 — APPROVED** [review]

**Review by ldewald — APPROVED** [review]

**Pull request merged**